shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Generator/Utxo.hs

```diff
@@ -372,22 +502,22 @@ calcOutputsFromBalance balance_ addrs fee =
 -- given UTxO originated from (in order to produce the appropriate witnesses to
 -- spend these outputs). If this is not the case, `findPayKeyPairAddr` /
 -- `findPayScriptFromAddr` will fail by not finding the matching keys or scripts.
-genInputs ::
-  (HasCallStack, Era era) =>
-  Constants ->
+genInputs :: forall v era.
```

Comment:
Again, this type signature is misleading. It claims to be a generator over an arbitrary value `v`, but it's not. The values that it generates are just `Coin`, albeit wrapped up in a `v`-shaped disguise. 

Better for the type signature to be clear about what it's generating. Then when we come to actually make the tests run over multiple eras, the type system guides us to which parts need changing.

Reply:
I don't agree. This is a case of anticipating possible changes, at no cost.